GetDKAN/dkan
# Clean up datastore controller and related classes

In preparation for new CSV pagination logic, organizing the datastore query controllers in a more straightforward way, to avoid bloat and code reuse. 

Code changes: 

- Introduction of an AbstractQueryController class, with the normal JSON queries and the streamed CSV queries both extending the same base class for consistency. SQL endpoint could later be brought in to follow this pattern, perhaps other datastore  variants in the future. Custom modules wanting to override or extend datastore behavior can also use this as a base.
- RequestParamNormalizer util class folded into AbstractQueryController
- Datastore routes consolidated to be easier to read
- CSV streaming code cleaned up to be more straightforward and easy to work with.
- The rows limit setting is now the query default as well as the maximum in the DatastoreQuery schema.

## QA Steps

- Set the rows limit to 1000 in admin/dkan/datastore-settings. Clear the cache.
- Confirm that https://dkan-qa-3696.ci.civicactions.net/api/1/datastore/query/schema reflects the limit setting in `['limit']` default and maximum
- Confirm that the limit has defaulted to the rows limit setting
- Other than this, the idea is this should not change anything for the user, for any exposed services or APIs. However, custom code should be checked as controllers and other classes have changed.
